Library/Homebrew/cmd/update-report.rb

```diff
@@ -253,6 +271,85 @@ def migrate_cache_entries_to_double_dashes(initial_version)
     end
   end
 
+  def migrate_cache_entries_to_symlinks(initial_version)
+    return if initial_version && initial_version > "1.7.2"
+
+    return if ENV.key?("HOMEBREW_DISABLE_LOAD_FORMULA")
+
+    ohai "Migrating cache entries..."
+
+    formula_downloaders = if HOMEBREW_CACHE.directory?
+      HOMEBREW_CACHE.children
+                    .select(&:file?)
+                    .map { |child| child.basename.to_s.sub(/\-\-.*/, "") }
+                    .uniq
+                    .map { |formula| Formula[formula] rescue nil } # rubocop:disable Style/RescueModifier
+                    .compact
+                    .flat_map { |formula| formula_resources(formula) }
+                    .map { |resource| [resource.downloader, resource.download_name, resource.version] }
+    else
+      []
+    end
+
+    cask_downloaders = if (HOMEBREW_CACHE/"Cask").directory?
+      (HOMEBREW_CACHE/"Cask").children
+                             .map { |child| child.basename.to_s.sub(/\-\-.*/, "") }
+                             .uniq
+                             .map { |cask| Hbc::CaskLoader.load(cask) rescue nil } # rubocop:disable Style/RescueModifier
```

Comment:
Can we make this into a bigger method with a proper rescue rather than needing the `rescue nil` here? This line is also too long.